mitodl/open-discussions
# Add stylelint for scss linting

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #3848 

#### What's this PR do?
This PR adds linting for `.scss` files in "new" open. Previous `.scss`-linting was only happening in `frontends/open-discussions`. Now it happens in all the `frontends/` packages:
- `frontends/open-discussions` continues to use the [`sass-lint`](https://github.com/sasstools/sass-lint) for scss linting
- All other packages in `frontends/` are using [`stylelint`](https://www.npmjs.com/package/stylelint)

The reason to use a new tool in the newer packages are primarily:
- [`sass-lint`](https://github.com/sasstools/sass-lint) is no longer maintained and hasn't been updated in 4 years
- **and in particular**, `sass-lint` does not support Sass module syntax (`@use`), which we have been using in infinite-corridor

#### How should this be manually tested?
On this branch:
1. Enter `watch` container terminal via something like `docker compose exec watch bash`. (Or prefix commands below with `docker compose run --rm watch <cmd>`)
2. Run `yarn scss-lint`. You should see each frontend package (the directories under `frontends/`) be linted. Linting should pass.
3. Introduce a linting error in any of the frontend packages other than `open-discussions`. E.g., add some extra newlines in `frontends/infinite-corridor/src/scss/learning-resources.scss`, or change a variable name from kebab-case to camelCase. Run `yarn scss-lint` again. It should fail with a helpful error message.
4. (Optional) If you use VS Code, try the [stylehint extension](https://marketplace.visualstudio.com/items?itemName=stylelint.vscode-stylelint). You'll need to add
    ```json
    {
            "stylelint.validate": ["css", "scss"]
    }
    ```
    to your `settings.json`. If you do, you should be able to see scss linting errors in the editing.
